### ISSUE TITLE
New style celery task jobs never enter queued state

### ISSUE THREAD

**Issue by danlamanna** [girder/girder_worker#220]
Job go immediately from inactive (the Girder default) to running. I have a local branch where I made use of the [after-task-publish](http://docs.celeryproject.org/en/latest/userguide/signals.html#after-task-publish) signal in Celery, but this seems susceptible to race conditions. On my local environment, the status is getting set to running by the prerun signal *before* the after task publish has gotten to set the status to queued.

**Comment by kotfic** [comment]
I don't believe there are any hooks between before_task_publish and when kombu puts the message on the queue.  I know it is possible/relatively easy to configure a custom kombu producer that could potentially extend the send_message function with a side-effect that updates the state before actually placing on the queue.  you (or someone) would have to investigate further though to see just how sticky that gets

**Comment by danlamanna** [comment]
One solution is to mark the job as queued before it's actually queued
(this is how old style tasks work). A downside for development is that
it would be counter-intuitive to debugging if the message were marked
"queued" without actually being put on the queue. Perhaps better logging
is part of the solution to this problem.

**Comment by kotfic** [comment]
ref: https://github.com/girder/girder_worker/issues/219

**Comment by zachmullen** [comment]
In addition to not entering the queued state, docker_run tasks that download girder inputs and upload girder outputs do not set the FETCHING_INPUT or PUSHING_OUTPUT flags as before.

It occurs to me that such a thing might be difficult under the new architecture, but if it's not too hard, I'd love that have this info back to easily gauge whether tasks are more IO bound or CPU bound.

**Comment by cjh1** [comment]
The sort of "custom" state transitions would now have to be done in the transforms I think, but we should think carefully about how they might interact with the cancelling functionality.

**Comment by zachmullen** [comment]
> we should think carefully about how they might interact with the cancelling functionality

What sort of trouble do you anticipate there?

**Comment by cjh1** [comment]
Not sure :-) We currently enforce state transitions [here](https://github.com/girder/girder/blob/master/plugins/jobs/server/constants.py#L71), would  need to ensure any state transition in transforms take this into account.

**Comment by kotfic** [comment]
just as a side note,  I am hoping that the state transition information will migrate into girder_worker_utils that way it can be used across girder & girder_worker. That would also make it easier to maintain custom states in common transforms (as those transforms will be kept in girder_worker_utils).
